lib/spack/spack/spec.py

```diff
@@ -1463,6 +1463,26 @@ def edges_to_dependencies(self, name=None, depflag: dt.DepFlag = dt.ALL):
         """
         return [d for d in self._dependencies.select(child=name, depflag=depflag)]
 
+    @property
+    def edge_attributes(self) -> str:
+        """Helper method to print edge attributes in spec literals"""
+        edges = self.edges_from_dependents()
+        if not edges:
+            return ""
+
+        union = DependencySpec(parent=Spec(), spec=self, depflag=0, virtuals=())
+        for edge in edges:
+            union.update_deptypes(edge.depflag)
+            union.update_virtuals(edge.virtuals)
+        deptypes_str = (
+            f"deptypes={','.join(dt.flag_to_tuple(union.depflag))}" if union.depflag else ""
+        )
+        virtuals_str = f"virtuals={','.join(union.virtuals)}" if union.virtuals else ""
+        if not deptypes_str and not virtuals_str:
+            return ""
+        result = f"{deptypes_str} {virtuals_str}".strip()
+        return f"[{result}]"
```

Comment:
I don't think this suggestion is quite correct. It looks to me like it would print `[" "]` if nothing needs to be printed, since using `" ".join` on two empty strings will create the non-empty string with a single space.

Reply:
Add `.strip()` then?  It's still clearer.